service/internal/configprovider/expand.go

```diff
@@ -15,21 +15,20 @@
 package configprovider // import "go.opentelemetry.io/collector/service/internal/configprovider"
 
 import (
+	"context"
 	"os"
 
 	"go.opentelemetry.io/collector/config"
 )
 
-// NewExpandConverter returns a service.ConfigMapConverterFunc, that expands all environment variables for a given config.Map.
-//
-// This does not directly return service.ConfigMapConverterFunc to avoid circular dependencies, not a problem since it is internal.
-func NewExpandConverter() func(*config.Map) error {
-	return func(cfgMap *config.Map) error {
+// NewExpandConverter returns a config.MapConverter, that expands all environment variables for a given config.Map.
+func NewExpandConverter() config.MapConverter {
```

Comment:
Please move this back to `configmapprovider`.  I don't see any reason for it to be internal and it is needed by downstream distributions.  Doing it here can save me the trouble of creating another PR to do it.

Reply:
This is a separate request, can you please open an issue. Trying to keep discussion focus on the change.

Reply:
So I can spend weeks chasing changes you're making while waiting for approval?  Or, you can `git mv` and insert `map` back into the package name while you're in the midst of making any semblance of a rationale for making this internal obsolete.

Reply:
Not going to argue more, but I can tell you that I will not mix that in the same PR since is a different change, I can do it after this PR.

I've learned a new English word "semblance":
> the outward appearance or apparent form of something, especially when the reality is different.